pkg/operator/encryption/state.go

```diff
@@ -74,16 +74,22 @@ func getDesiredEncryptionState(encryptionConfig *apiserverconfigv1.EncryptionCon
 	// this is our output from the for loop below
 	encryptionState := groupResourcesState{}
 
-	// TODO if we cannot find the secrets associated with the keys, they may have been deleted
-	// we could try to use the key data in the revision to recover or just go degraded permanently
 	for gr, keys := range getGRsActualKeys(encryptionConfig) {
-		writeSecret := findSecretForKey(keys.writeKey, encryptionSecrets, targetNamespace) // TODO handle nil as error when hasWriteKey == true?
+		writeSecret := findSecretForKey(keys.writeKey, encryptionSecrets, targetNamespace)
```

Comment:
wouldn't returning an error when a key hasn't been found make more sense?

Reply:
The caller has to determine if not finding a key is an error though.  At most I would expect `findSecretForKey` to return a `bool` to indicate if it found it, but the `nil` check seems enough.